android/src/main/kotlin/com/airship/flutter/ConfigManager.kt

```diff
@@ -27,41 +41,178 @@ class ConfigManager(private val context: Context) {
+                Pair(preferences[APP_KEY], preferences[APP_SECRET])
+            }.first()
+        context.flutterPluginStore.updateData {
+            Config.AirshipConfig.parseFrom(configByteArray).apply {
+                // if defaultEnv appKey/secretKey has been left blank in dart config,
+                // and there is app credentials stored in preferences;
+                // the stored credentials are used to override defaultEnv
+                if (defaultEnv.isSet) {
+                    if (!(appCredentialsOverride.first.isNullOrEmpty() || appCredentialsOverride.second.isNullOrEmpty())) {
+                        this.defaultEnv.toBuilder().apply {
+                            appKey = appCredentialsOverride.first
+                            appSecret = appCredentialsOverride.second
+                        }.build()
+                    }
+                }
+            }
         }
     }
 
     val config: Flow<AirshipConfigOptions>
         get() = flow {
-            val appCredentialsOverride =
-                context.airshipFlutterPluginStore.data.map { preferences ->
-                    Pair(preferences[APP_KEY], preferences[APP_SECRET])
-                }.first()
+            /// Plugin config stored/ from dart config
+            val pluginConfig = context.flutterPluginStore.data.first()
 
             val config = AirshipConfigOptions.newBuilder()
                 .applyDefaultProperties(context)
+                .setRequireInitialRemoteConfigEnabled(true)
                 .apply {
-                    if (!(appCredentialsOverride.first.isNullOrEmpty() || appCredentialsOverride.second.isNullOrEmpty())) {
-                        this.setAppKey(appCredentialsOverride.first)
-                            .setAppSecret(appCredentialsOverride.second)
+                    /// since proto buf defaults to empty string,
+                    // check for empty string instead of null
+                    pluginConfig.defaultEnv.let {
+                        this.setAppKey(it.appKey)
+                            .setAppSecret(it.appSecret)
+                            .setLogLevel(it.logLevel.value)
+                            .build()
+                    }
+
+                    pluginConfig.development.let {
+                        this.setDevelopmentAppKey(it.appKey)
+                            .setDevelopmentAppSecret(it.appSecret)
+                            .setDevelopmentLogLevel(it.logLevel.value)
                             .build()
                     }
+                    pluginConfig.production.let {
+                        this.setProductionAppKey(it.appKey)
+                            .setProductionAppSecret(it.appSecret)
+                            .setProductionLogLevel(it.logLevel.value)
+                            .build()
+                    }
+
+
+                    this.setSite(
+                        when (pluginConfig.site!!) {
```

Comment:
can we mark this as nonnull somewhere else? maybe fallback to US if not set? we usually avoid force unwrapping when possible